src/figuresClasses.ts

```diff
@@ -1,11 +1,70 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
+  ) {
+    const sideOfTriangle = [a, b, c].sort((a1, b1) => b1 - a1);
+    const condition =
+      sideOfTriangle[0] >= sideOfTriangle[1] + sideOfTriangle[2];
+
+    if (a <= 0 || b <= 0 || c <= 0 || condition) {
+      throw new Error('your error message');
```

Comment:
The error message thrown should contain a detailed description of the error to provide clarity on what went wrong.